google/cloud/aiplatform/utils.py

```diff
@@ -296,20 +306,29 @@ def extract_bucket_and_prefix_from_gcs_path(gcs_path: str) -> Tuple[str, Optiona
     return (gcs_bucket, gcs_blob_prefix)
 
 
-class WrappedClient:
-    """Wrapper class for client that creates client at API invocation time."""
+class ClientWithOverride(metaclass=abc.ABCMeta):
+    """Wrapper class for client that creates client with override at API invocation time."""
+
+    @property
+    @classmethod
+    @abc.abstractmethod
+    def _default_version(self) -> str:
+        pass
+
+    @property
+    @classmethod
+    @abc.abstractmethod
+    def _version_map(self) -> Tuple:
+        pass
 
     def __init__(
```

Comment:
We should provide the option to not call through the temporary client for prediction because it is latency sensitive.  The lines for checking were removed from the `_Config.create_client`. 

I think this can done in a few ways. One option is to nest the WrappedClient class in this class like this and construct the client at instantiation:

```python
class ClientWithOverride:
  class WrappedClient:
    """Wrapper class for client that creates client at API invocation time."""

    def __init__(
        self,
        client_class: Type[AiPlatformServiceClient],
        client_options: client_options.ClientOptions,
        client_info: gapic_v1.client_info.ClientInfo,
        credentials: Optional[auth_credentials.Credentials] = None,
    ):
        self._client_class = client_class
        self._credentials = credentials
        self._client_options = client_options
        self._client_info = client_info

    def __getattr__(self, name: str) -> Any:
        """Instantiates client and returns attribute of the client."""
        temporary_client = self._client_class(
            credentials=self._credentials,
            client_options=self._client_options,
            client_info=self._client_info,
        )
        return getattr(temporary_client, name)

  @abstract class property
  _is_temporary: Bool

  def __init__(
      self, credentials, client_options,  client_info):
      self._clients = {
          version: WrappedClient(client, credentials, client_options, client_info) if self._is_temporary else client(...)
          for version, client in self._version_map
      }

  def __getattr__(self, name: str) -> Any:
      """Instantiates client and returns attribute of the client."""
      return getattr(self._clients[self._default_version], name)

  def select_version(self, version: str) -> AiPlatformServiceClient:
      return self._clients[version]
```

The main goal is to avoid the additional check to see if we should use the temporary client at every API invocation.